src/store/functions/getToken.ts

```diff
@@ -1,12 +1,28 @@
-import axios from "axios";
 import { getURLCode } from ".";
+import { useTokenStore, axiosPTO } from "..";
+import { getNewToken } from ".";
 
 export async function getToken() {
   try {
     const code = getURLCode();
     if (code) {
-      const res = await axios.post(
-        `${process.env.NEXT_PUBLIC_API_BASE_URL}/token`,
+      // check for existing token
+      const { accessToken: currentToken, accessTokenExpiresAt } =
+        useTokenStore.getState();
+      if (currentToken) {
+        const now = new Date();
+        const expiresAt = new Date(accessTokenExpiresAt);
+        if (now < expiresAt) {
+          console.log("token still valid, not getting new one");
```

Comment:
Best practice is to not leave `console.log`s in final code.

Reply:
removed 'em. thanks